### PR TITLE
added save function to enum_plot_panel

### PR DESCRIPTION
**Pull request by LisaSchulz97**
the value of the enum_panels getting saved now

### CODE REVIEW

File: tools/twix/src/panels/enum_plot.rs [hunk 1]
```diff
@@ -97,17 +97,31 @@ pub struct EnumPlotPanel {
 impl Panel for EnumPlotPanel {
     const NAME: &'static str = "Enum Plot";
 
-    fn new(nao: Arc<Nao>, _value: Option<&serde_json::Value>) -> Self {
-        Self {
+    fn new(nao: Arc<Nao>, value: Option<&serde_json::Value>) -> Self {
```
[the remark this answers is not in the document]
Comment (LisaSchulz97) [reply tools/twix/src/panels/enum_plot.rs]: done

File: tools/twix/src/panels/enum_plot.rs [hunk 2]
```diff
@@ -97,17 +97,31 @@ pub struct EnumPlotPanel {
...
+
+        let mut panel = Self {
             nao,
             changes: Vec::new(),
             messages_count: 0,
             change_buffer: None,
             scroll_position: 0.0,
             viewport_width: 1.0,
-            output_key: String::new(),
+            output_key: output,
```
[the remark this answers is not in the document]
Comment (LisaSchulz97) [reply tools/twix/src/panels/enum_plot.rs]: done

File: tools/twix/src/panels/enum_plot.rs [hunk 3]
```diff
@@ -97,17 +97,27 @@ pub struct EnumPlotPanel {
 impl Panel for EnumPlotPanel {
     const NAME: &'static str = "Enum Plot";
 
-    fn new(nao: Arc<Nao>, _value: Option<&serde_json::Value>) -> Self {
-        Self {
+    fn new(nao: Arc<Nao>, value: Option<&Value>) -> Self {
+        let output_key = value.and_then(|value| String::from(value.get("subscribe_key").as_str())).unwrap_or_default();
```
Comment (Narcha) [inline tools/twix/src/panels/enum_plot.rs:101]: whoops, this should do it
```suggestion
        let output_key = value
            .and_then(|value| value.get("subscribe_key"))
            .and_then(|value| value.as_str())
            .map(String::from)
            .unwrap_or_default();

```

File: tools/twix/src/panels/enum_plot.rs [hunk 4]
```diff
@@ -97,17 +97,31 @@ pub struct EnumPlotPanel {
...
             viewport_mode: ViewportMode::Full,
-        }
+        };
+        panel.subscribe();
+        panel
+    }
+
+    fn save(&self) -> Value {
+        json!({
+            "subscribe_key": self.output_key.clone()
```
Comment (Narcha) [inline tools/twix/src/panels/enum_plot.rs:123]: Remove unnecessary clone
```suggestion
            "subscribe_key": self.output_key
```

**Review by Narcha — APPROVED** [review]
LGTM :)

**Pull request merged**